ghengin-core/ghengin-core/Ghengin/Core/Render/Property.hs

```diff
@@ -26,6 +27,8 @@ import qualified Data.IntMap.Linear as IM
 
 import qualified Vulkan as Vk -- TODO: Core shouldn't depend on any specific renderer implementation external to Core
 
+data BufferType = Uniform | Storage
```

Comment:
This is an implementation detail of Vulkan. Should we use some generic type for this, like LargeBuffer and SmallBuffer? That seems awfully confusing.